### PR TITLE
platform agnostic lb ep hash

### PR DESCRIPTION
**Pull request by stevenctl**
# Description

  Don't use native endian-ness when generating hashes.
      This creates difficulty reproducing tests on different machines
      where the hashes end up in goldenfiles.

### CODE REVIEW

File: projects/gateway2/krtcollections/endpoints_test.go [hunk 1]
```diff
@@ -235,11 +236,23 @@ func TestEndpointsForUpstreamWithDiscoveredUpstream(t *testing.T) {
...
 
-	h1 := result1.LbEpsEqualityHash ^ result2.LbEpsEqualityHash
-	h2 := result3.LbEpsEqualityHash ^ result4.LbEpsEqualityHash
+	h1 := combineHashes(result1.LbEpsEqualityHash, result2.LbEpsEqualityHash)
+	h2 := combineHashes(result3.LbEpsEqualityHash, result4.LbEpsEqualityHash)
 
 	g.Expect(h1).NotTo(Equal(h2), "not expected %v, got %v", h1, h2)
+}
+
+func combineHashes(hash1, hash2 uint64) uint64 {
```
Comment (yuval-k) [inline projects/gateway2/krtcollections/endpoints_test.go:245]: why this change?

**Review by yuval-k — CHANGES_REQUESTED** [review]
also, there's a change needed  in `projects/gateway2/utils/hash.go` in HashUint64

File: projects/gateway2/proxy_syncer/cla.go [hunk 2]
```diff
@@ -129,18 +130,23 @@ func NewPerClientEnvoyEndpoints(logger *zap.Logger, dbg *krt.DebugHandler, uccs
...
 	var priorityInfo *PriorityInfo
 
+	updatedHash := ep.LbEpsEqualityHash
 	if destrule != nil {
 		trafficPolicy := getTraficPolicy(destrule, ep.Port)
 		localityLb := getLocalityLbSetting(trafficPolicy)
 		if localityLb != nil {
 			priorityInfo = getPriorityInfoFromDestrule(localityLb)
 			hasher := fnv.New64()
+
```
[the remark this answers is not in the document]
Comment (stevenctl) [reply projects/gateway2/proxy_syncer/cla.go:142]: reverting it

File: projects/gateway2/proxy_syncer/cla.go [hunk 3]
```diff
@@ -153,7 +153,7 @@ func PrioritizeEndpoints(logger *zap.Logger, destrule *DestinationRuleWrapper, e
 	return UccWithEndpoints{
 		Client:        ucc,
 		Endpoints:     resource.NewEnvoyResource(cla),
-		EndpointsHash: ep.LbEpsEqualityHash ^ additionalHash,
```
Comment (yuval-k) [inline projects/gateway2/proxy_syncer/cla.go:156]: revert this change?

**Review by yuval-k — APPROVED** [review]

File: changelog/v1.18.0-rc3/platform-agnostic-lbephash.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+changelog:
+  - type: NON_USER_FACING
+    description: >-
+      Don't use xor or native endian-ness when generating hashes.
```
Comment (lgadban) [inline changelog/v1.18.0-rc3/platform-agnostic-lbephash.yaml:4]: can we update changelog (and PR description) to reflect that we are keeping xor ?

**Pull request merged**